src/deblock.rs

```diff
@@ -0,0 +1,439 @@
+                 blimit: u16, limit: u16, thresh: u16, bd: usize) {
+    let mut s = 0;
+    let flat = 1 << bd - 8;
+    let data = slice.as_mut_slice();
+    for _i in 0..4 {
+        let p = &mut data[s..];
+        let p2 = p[0] as i32;
+        let p1 = p[pitch] as i32;
+        let p0 = p[pitch*2] as i32;
+        let q0 = p[pitch*3] as i32;
+        let q1 = p[pitch*4] as i32;
+        let q2 = p[pitch*5] as i32;
+        if (p2 - p1).abs() as u16 <= limit &&
+            (p1 - p0).abs() as u16 <= limit &&
+            (q1 - q0).abs() as u16 <= limit &&
+            (q2 - q1).abs() as u16 <= limit &&
+            (p0 - q0).abs() as u16 * 2 + (p1 - q1).abs() as u16 / 2 <= blimit {
+                if (p1 - p0).abs() as u16 <= flat &&
+                    (q1 - q0).abs() as u16 <= flat &&
+                    (p2 - p0).abs() as u16 <= flat &&
+                    (q2 - q0).abs() as u16 <= flat {
+                        deblock6(p, pitch, p2, p1, p0, q0, q1, q2);
+                    } else {
+                        deblock4(thresh, bd, &mut p[pitch..], pitch, p1, p0, q0, q1);
+                    }
+            }
+        s += stride;
+    }
+}
+
+fn deblock_edge8<'a>(slice: &'a mut PlaneMutSlice<'a>, pitch: usize, stride: usize,
+                 blimit: u16, limit: u16, thresh: u16, bd: usize) {
+    let mut s = 0;
+    let flat = 1 << bd - 8;
+    let data = slice.as_mut_slice();
+    for _i in 0..4 {
+        let p = &mut data[s..];
+        let p3 = p[0] as i32;
+        let p2 = p[pitch] as i32;
+        let p1 = p[pitch*2] as i32;
+        let p0 = p[pitch*3] as i32;
+        let q0 = p[pitch*4] as i32;
+        let q1 = p[pitch*5] as i32;
+        let q2 = p[pitch*6] as i32;
+        let q3 = p[pitch*7] as i32;
+        if (p3 - p2).abs() as u16 <= limit &&
+            (p2 - p1).abs() as u16 <= limit &&
+            (p1 - p0).abs() as u16 <= limit &&
+            (q1 - q0).abs() as u16 <= limit &&
+            (q2 - q1).abs() as u16 <= limit &&
+            (q3 - q2).abs() as u16 <= limit &&
+            (p0 - q0).abs() as u16 * 2 + (p1 - q1).abs() as u16 / 2 <= blimit {
+                if (p1 - p0).abs() as u16 <= flat &&
+                    (q1 - q0).abs() as u16 <= flat &&
+                    (p2 - p0).abs() as u16 <= flat &&
+                    (q2 - q0).abs() as u16 <= flat &&
+                    (p3 - p0).abs() as u16 <= flat &&
+                    (q3 - q0).abs() as u16 <= flat {
+                        deblock8(p, pitch, p3, p2, p1, p0, q0, q1, q2, q3);
+                    } else {
+                        deblock4(thresh, bd, &mut p[pitch*2..], pitch, p1, p0, q0, q1);
+                    }
+            }
+        s += stride;
+    }
+}
+
+fn deblock_edge14<'a>(slice: &'a mut PlaneMutSlice<'a>, pitch: usize, stride: usize,
+                 blimit: u16, limit: u16, thresh: u16, bd: usize) {
+    let mut s = 0;
+    let flat = 1 << bd - 8;
+    let data = slice.as_mut_slice();
+    for _i in 0..4 {
```

Comment:
Same as above.

Reply:
thanks!  I'll roll them in, with a few other changes I want to make, on monday.